From 3aa6dbe257b5ebf21e40739abcfa456df95b30c5 Mon Sep 17 00:00:00 2001
From: Eddyhan <eddyhan@ami.com>
Date: Wed, 28 Sep 2022 15:47:42 +0800
Subject: [PATCH] Fix build issue.


diff --git a/CMakeLists.txt b/CMakeLists.txt
index df88a05d..6e93764b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,7 +152,7 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 
     if(TOOLCHAIN STREQUAL "GCC")
         SET(CMAKE_C_COMPILER gcc)
-        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -DUSING_LTO  -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare -Werror-implicit-function-declaration -Wcast-qual)
+        ADD_COMPILE_OPTIONS(-std=c99 -fshort-wchar -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fpie -fno-asynchronous-unwind-tables -DUSING_LTO  -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare -Werror-implicit-function-declaration -Wcast-qual -Wno-error=cast-qual)
         if (ARCH STREQUAL "x64")
             ADD_COMPILE_OPTIONS(-mno-red-zone)
         endif()
-- 
2.25.1

